mypy/plugin.py

```diff
@@ -351,6 +351,15 @@ def final_iteration(self) -> bool:
         raise NotImplementedError
 
 
+# A context for querying for configuration data about a module for
+# cache invalidation purposes.
+ReportConfigContext = NamedTuple(
+    'DynamicClassDefContext', [
+        ('id', str),        # Module name
+        ('path', str),      # Module file path
+        ('is_check', bool)  # Is this invocation for checking whether the config matches
```

Comment:
I still think it is not the best name, because it explains for what _we_ need this, while a plugin writer may not be fluent in mypy internals. I would say `on_load` or `on_loading` or similar would be already better. But up to you.

Maybe add a longer comment explaining that this hook is called twice: on loading cache and on storing cache.

Reply:
The fact that we are loading doesn't seem like the most relevant part to me, since the hook isn't doing any loading, while it might do some checking